breadcord/config.py

```diff
@@ -73,10 +74,12 @@ def root(self) -> SettingsGroup:
         node = self
         while node.parent is not None:
             node = node.parent
-        return node
+        if __debug__ and not isinstance(node, SettingsGroup):
```

Comment:
This feels a bit questionable, and I'd like some feedback on what we think of it

Reply:
Because I'm braindead right now, what does this achieve?

Reply:
It makes the type checker happy be eliminating the case where the root is not a settings group. This should never happen and I’ve hence only made it run in debug

Reply:
Might be worth adding a comment stating that it's an impossible branch then, otherwise would cause confusion when reading ("how could I end up in a state where this isn't SettingsGroup?")

Reply:
Maybe (I say, knowing full and well I will forget about this in 5 minutes)